src/Microsoft.OpenApi.Readers/V2/OpenApiOperationDeserializer.cs

```diff
@@ -136,6 +136,9 @@ internal static OpenApiOperation LoadOperation(ParseNode node)
                 ProcessProduces(node.CheckMapNode("responses"), response, node.Context);
             }
 
+            // Reset so that it's not picked up later
+            node.Context.SetTempStorage(TempStorageKeys.OperationProduces, null);
```

Comment:
Is this needed? It's already set in line 110 above

Reply:
Yes, to clear it our after we exit operation scope. So that when we process global "produces" later on we do not pickup ones stored in temp operation store.

Reply:
Got it. I'll investigate more on what to do with line 110, but this should not block this from being checked in.

Reply:
Both of them are needed. I'd clean other temp storages too (body, form, consumes), but don't want to increase scope of this PR. The general idea is to clean temp storage before operation is parsed, since parsing may read data from there and we don't want to use other operation's data. And we need to clean after as well if we're going to reuse same parsing logic.

In this particular test - two responses - without cleaning temp storage POST operation was being assigned two media types since they're left in temp storage after last operation and ProcessProduces prefers operation data to global.

Another approach here is to store data with scope, i.e. at line 65:

`n.Context.SetTempStorage(TempStorageKeys.OperationProduces, null, o)`

And pass operation reference to ProcessProduces and do:

`context.GetFromTempStorage<List<string>>(TempStorageKeys.OperationProduces, operation)`

This way it also would not be possible for one operation to be contaminated with other operation data. But I'm reluctant to make that change since it requires more thoroughal testing than I can do at the moment.

Reply:
The issue is that there are ProcessProduces in two places - in the operation and document deserializers. This leads to the data in temp storage being used incorrectly when ProcessProduces in called in document deserializer.

A couple more alternatives: 

- We technically do not need to clear OperationProduces in LoadOperation I think. We can also do it before we call ProcessProduces in the document deserializer (the same way we call it in line 110 of the operation deserializer here).

- We can make ProcessProduces take another parameter whether it's called from document or operation deserializer. Then, we only use the global produces when it's called from the document. 

- The scope method you propose should work as well with the right implementation.

Reply:
- We could clear it out before calling ProcessProduces at document level, but that would mean "exposing" internal dependencies of ProcessProduces.

- Would work

- Support for scope is already there, was added before for some other place. Technically should be used throughout, but it's a more broad change.